sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -1321,3 +1320,39 @@ def augment_options(cls, options):
       for name, value in override.items():
         setattr(options, name, value)
     return options
+
+
+class S3Options(PipelineOptions):
+  @classmethod
+  def _add_argparse_args(cls, parser):
+    # These options are passed to the S3 IO Client
+    parser.add_argument(
+        '--aws_access_key_id',
+        default=None,
+        help='The secret key to use when creating the s3 client.')
+    parser.add_argument(
+        '--aws_secret_access_key',
+        default=None,
+        help='The secret key to use when creating the s3 client.')
+    parser.add_argument(
+        '--aws_session_token',
+        default=None,
+        help='The session token to use when creating the s3 client.')
+    parser.add_argument(
+        '--endpoint_url',
+        default=None,
+        help='The complete URL to use for the constructed s3 client.')
+    parser.add_argument(
+        '--region_name',
+        default=None,
+        help='The name of the region associated with the client.')
+    parser.add_argument(
+        '--api_version', default=None, help='The API version to use.')
+    parser.add_argument(
+        '--verify',
+        default=None,
+        help='Whether or not to verify SSL certificates.')
+    parser.add_argument(
+        '--use_ssl',
+        default=True,
+        help='Whether or not to use SSL. By default, SSL is used.')
```

Comment:
You are correct that it's desirable to use a sort of namespace prefix. Perhaps `--aws_` or `--aws_s3_`. What do you think?

You must know more than me about s3 and AWS - I wonder if `aws_session_token`, `aws_secret_access_key`, `aws_access_key_id`  in this context are specific for s3, or if they provide some sort of AWS-wide authentication?

If they're s3-specific, then maybe we should namespace them as `aws_s3`? Let me know what you think.

Reply:
+1. Please do not use endpoint_url, region_name, api_version, verify, use_ssl and so on without a prefix.

Reply:
Fair enough. I've moved them all to use the same s3 prefix which I think makes sense given they are collected under the S3Options class. @pabloem I believe the access keys can also be used for other AWS services, although I've never actually used them. I think it makes sense to consolidate the more generic aws options with the s3 options together for now given that this is the only use case at the moment. If there is another AWS service added in the future it could make sense then to split them up.

Reply:
Thanks! That makes sense to me. Can you fix the broken unittests? Other than that, the change looks great (and it's very welcome, as we'd been needing it).

Reply:
Will do. I tried to have a look at the failing tests and can't figure out which ones have actually failed (the formatting is quite difficult to parse). Unfortunately I don't have access to windows to run locally. The same pattern of failures seems to be affecting https://github.com/apache/beam/pull/13187 which is just a comment change, so perhaps the failures are unrelated?